src/tls.cpp

```diff
@@ -142,192 +145,47 @@ static int glibcMinorVersion()
-     * Then fill in the entry_number field of an empty 'struct user_desc', and
-     * get_thread_area(struct user_desc *uinfo) will fill in the rest.
-     * Then use the filled in base_address field to get the 'struct pthread'.
-     * The function get_tls_base_addr() returns this 'struct pthread' addr.
-     */
-    void *pthread_desc = (void*) pthread_self();
-
-    /* A false hit for tid_offset probably can't happen since a new
-     * 'struct pthread' is zeroed out before adding tid and pid.
-     * pthread_desc below is defined as 'struct pthread' in glibc:nptl/descr.h
-     */
-    tmp = memsubarray((char *)pthread_desc, (char *)&tid_pid, sizeof(tid_pid));
-
-    if (tmp == NULL && glibcMajorVersion() == 2 && glibcMinorVersion() >= 24) {
-      // starting with glibc-2.25 (including 2.24.90 on Fedora), the pid field
-      // is deprecated and set to zero.
-      tid_pid.pid = 0;
-      tmp = memsubarray((char*)pthread_desc, (char *)&tid_pid, sizeof(tid_pid));
-    }
-
-    if (tmp == NULL) {
-      JWARNING(false) (tid_pid.tid)
-        .Text("Couldn't find offsets of tid/pid in thread_area.\n"
-              "  Now relying on the value determined using the\n"
-              "  glibc version with which DMTCP was compiled.");
-      return STATIC_TLS_TID_OFFSET();
-
-      // mtcp_abort();
-    }
-
-    tid_offset = tmp - (char *)pthread_desc;
-    JWARNING (tid_offset == STATIC_TLS_TID_OFFSET()) (tid_offset)
-      .Text("tid_offset (%d) different from expected.\n"
-             "  It is possible that DMTCP was compiled with a different\n"
-             "  glibc version than the one it's dynamically linking to.\n"
-             "  Continuing anyway.  If this fails, please try again.");
-
-    if (tid_offset % sizeof(int) != 0) {
-      JWARNING(tid_offset %sizeof(int) == 0) (tid_offset)
-        .Text("tid_offset is not divisible by sizeof(int).\n"
-             "  Now relying on the value determined using the\n"
-             "  glibc version with which DMTCP was compiled.");
-      return tid_offset = STATIC_TLS_TID_OFFSET();
-
-      // mtcp_abort();
-    }
-
-    /* Should we do a double-check, and spawn a new thread and see
-     *  if its TID matches at this tid_offset?  This would give greater
-     *  confidence, but for the reasons above, it's probably not necessary.
-     */
-  }
-  return tid_offset;
-}
-
 /*****************************************************************************
  *
  *****************************************************************************/
 int
 TLSInfo_GetPidOffset(void)
 {
-  static int pid_offset = -1;
+  static int pid_offset = TLSInfo_GetTidOffset() + sizeof(pid_t);
 
-  struct { pid_t tid; pid_t pid; } tid_pid;
-  if (pid_offset == -1) {
-    int tid_offset = TLSInfo_GetTidOffset();
-    pid_offset = tid_offset + (char *)&(tid_pid.pid) - (char *)&tid_pid;
-    JTRACE("") (pid_offset);
-  }
   return pid_offset;
 }
 
-static char *
-memsubarray(char *array, char *subarray, size_t len)
```

Comment:
This seems like a utility function we could use. Instead of removing it, can we move it either in mtcp or util?

Reply:
We can use std::find or friends if we need similar functionality later on.